fedora/infra.py

```diff
@@ -150,31 +197,119 @@ def format_title(outage):
+            await evt.respond(message)
+            return
+
+        message += (
+            f"**{len(ongoing)} ongoing** and **{len(planned)} planned** "
+            f"outages on Fedora Infrastructure.{NL}"
+        )
+
+        if ongoing:
+            message += f"##### Ongoing{NL}"
+            for outage in ongoing:
+                start_time = self.date_validation(outage.get("startdate"))
+                end_time = (
+                    self.date_validation(outage.get("enddate"))
+                    if outage.get("enddate") is not None
+                    else None
+                )
+
+                now = datetime.datetime.now(start_time.tzinfo)
+                delta_time = now - start_time
+
+                relative_start_time = self.get_relative_time(delta_time)
+                relative_end_time = (
+                    self.get_relative_time(end_time - now) if end_time is not None else None
+                )
+
+                message += f" * {format_title(outage)}{NL}"
+                message += f"   Started : {relative_start_time}{NL}{start_time}"
+
+                message += (
+                    f"   Estimated to end: "
+                    f"{end_time if end_time else 'Unknown'}{NL}"
+                    f"{relative_end_time  if relative_end_time else 'Unknown'}{NL}"
+                )
+
             if planned:
-                message = message + f"##### Planned{NL}"
+                message += f"##### Planned{NL}"
                 for outage in planned:
-                    message = message + f" * {format_title(outage)}{NL}"
-                    # TODO: Make these times relative (e.g. started 1 hour ago)
-                    message = message + f"   Scheduled to start at: {outage['startdate']}{NL}"
-                    message = message + (
-                        f"   Scheduled to end at: "
-                        f"{outage['enddate'] if outage['enddate'] else 'Unknown'}{NL}"
+                    message += f" * {format_title(outage)}{NL}"
+
+                    start_time = self.date_validation(outage.get("startdate"))
+                    relative_start_time = self.get_relative_time(start_time - now)
+
+                    message += (
+                        f"   Scheduled to start :" + f"{start_time}" + f"{relative_start_time}{NL}"
                     )
+                    message += (
+                        f"   Estimated to end: "
+                        f"{outage.get('enddate') if outage.get('enddate') else 'Unknown'}{NL}"
+                        f"{self.get_relative_time(outage.get('enddate') - now  if outage.get('enddate') else datetime.datetime.max - now)}{NL}"
+                    )
+
+            await evt.respond(message)
+
+    def date_validation(self, date_string):
+        date_format = "%Y-%m-%dT%H:%M:%S%z"
+        date_string = datetime.datetime.strptime(date_string, date_format)
+        return date_string
+
+    def get_relative_time(self, delta):
```

Comment:
It looks like the project already has a dependency on the `arrow` library, which provides a [humanize](https://arrow.readthedocs.io/en/latest/api-guide.html#arrow.arrow.Arrow.humanize) call to calculate relative times. I'd recommend using that instead of implementing it yourself (although your implementation looks nice and simple!)

Reply:
> It looks like the project already has a dependency on the `arrow` library, which provides a [humanize](https://arrow.readthedocs.io/en/latest/api-guide.html#arrow.arrow.Arrow.humanize) call to calculate relative times. I'd recommend using that instead of implementing it yourself (although your implementation looks nice and simple!)

*inserts crying emoji* I knew there was something i could use but the DSA warrior in me said no. It took a while to come up with the logic hahah. Thank you for the review!